share/ansible_navigator/utils/image_introspect.py

```diff
@@ -0,0 +1,296 @@
+""" introspect an image """
+import json
+import re
+import subprocess
+import sys
+import multiprocessing
+from types import SimpleNamespace
+
+from typing import Any
+from typing import Callable, List
+from typing import Dict
+from typing import Union
+from queue import Queue
+
+# pylint: disable=broad-except
+
+PROCESSES = (multiprocessing.cpu_count() - 1) or 1
+
+
+class Command(SimpleNamespace):
+    """command obj holder"""
+
+    id: str
+    command: str
+    parse: Callable
+    stdout: str = ""
+    stderr: str = ""
+    details: Union[List, Dict, str] = ""
+    errors: List = []
+
+
+def run_command(command: Command) -> None:
+    """run a command"""
+    try:
+        proc_out = subprocess.run(
+            command.command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=True,
+            universal_newlines=True,
+            shell=True,
+        )
+        command.stdout = proc_out.stdout
+    except subprocess.CalledProcessError as exc:
+        command.stderr = str(exc.stderr)
+        command.errors = [str(exc.stderr)]
+
+
+def worker(pending_queue: multiprocessing.Queue, completed_queue: multiprocessing.Queue) -> None:
+    """a worker, pulls from pending, runs & porcesses
+    places in completed"""
+    while True:
+        command = pending_queue.get()
+        if command is None:
+            break
+        run_command(command)
+        try:
+            command.parse(command)
+        except Exception as exc:
+            command.errors = command.errors + [str(exc)]
+        completed_queue.put(command)
+
+
+class CommandRunner:
```

Comment:
Since `ansible-runner` will be part of EE can this be handled directly by invoking the runner API's? In future when runner supports all the ansible API's (eg. ansible-galaxy) it will be easier to refactor if we have a single entry point to run Ansible as well as generic commands

Reply:
See #343